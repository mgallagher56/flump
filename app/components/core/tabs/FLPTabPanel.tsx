import React from 'react';


const FLPTabPanel: React.FC<React.HTMLAttributes<HTMLElement>> = (props) => {
  return <div {...props}>{props.children}</div>;
};

export default FLPTabPanel;
