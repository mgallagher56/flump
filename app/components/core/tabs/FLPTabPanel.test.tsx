import type { ReactElement, ReactNode } from 'react';

import { Tabs } from '@chakra-ui/react';
import { render } from '@testing-library/react';

import FLPTabPanel from './FLPTabPanel';

const tabWrapper = (children: ReactNode): ReactElement => <Tabs>{children}</Tabs>;

describe('FLPTabPanel', () => {
  test('renders correctly', () => {
    const { container } = render(
      tabWrapper(
        <>
          <FLPTabPanel>
            <div>Tab content 1</div>
          </FLPTabPanel>
          <FLPTabPanel>
            <div>Tab content 2</div>
          </FLPTabPanel>
          <FLPTabPanel>
            <div>Tab content 3</div>
          </FLPTabPanel>
        </>
      )
    );
    expect(container).toMatchSnapshot();
  });
});
