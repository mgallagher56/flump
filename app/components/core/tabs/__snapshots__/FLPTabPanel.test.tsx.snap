// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPTabPanel > renders correctly 1`] = `
<div>
  <div
    class="chakra-tabs css-0"
  >
    <div>
      <div>
        Tab content 1
      </div>
    </div>
    <div>
      <div>
        Tab content 2
      </div>
    </div>
    <div>
      <div>
        Tab content 3
      </div>
    </div>
  </div>
</div>
`;
